### PR TITLE
fix: [CI-5116]: MultiTypeMapInputSet allows empty run pipeline and pe…

### PR DESCRIPTION
**Pull request by mtran7**
…rsist empty values inputset

### Summary
**Read this first**
https://harness.atlassian.net/wiki/spaces/PD/pages/21180449611/MultiTypeInputSetMap+Component

Many deleted lines because we no longer render default placeholder key/value row
image

`ui-pipeline` @vkbansal Change in InputSetSelector for `setSelectedInputSetsContext` was necessary because previously Applying Input Sets on Run Pipeline Modal did not re-render the map component. We had to switch to YAML view and back to Visual to see actual values. Re-rendering map component is tricky because we do not want to re-render after every key-press since the user will lose focus while typing. 

`ui-sto` required because one snapshot was updated where we no longer pre-populate the map component w/ 1 row of key/value pairs. 

Additional Fixes
1. New onEdit Input Set Page flakiness problem was introduced for map variables (envVariables) but taken care of by re-rendering for `mapFieldsOverwritten` and `registeredPrefixes`. 

Test Cases here have passed: https://docs.google.com/spreadsheets/d/1Z7DxBDLVOa5UGgPX0ipy_99gt7evD3h46_Y_o1GLfzs/edit#gid=0

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

## [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11126)

File: src/modules/70-pipeline/components/PipelineInputSetForm/__tests__/StageInputSetForm.test.tsx [hunk 1]
```diff
@@ -369,7 +369,8 @@ describe('stageinputset tests', () => {
       )
       const deleteButton = container.querySelectorAll('[data-icon="main-trash"]')
       fireEvent.click(deleteButton[0])
-      expect(container.querySelectorAll('[data-icon="main-trash"]').length).toBe(3)
+
+      expect(container.querySelectorAll('[data-icon="main-trash"]').length).toBe(1)
```
Comment (mtran7) [inline src/modules/70-pipeline/components/PipelineInputSetForm/__tests__/StageInputSetForm.test.tsx:373]: we are no longer defaulting map components to have 1 placeholder row. Therefore the trash icons won't be there

File: src/modules/75-ci/components/PipelineSteps/CIStep/CIStepOptionalConfig.tsx [hunk 2]
```diff
@@ -543,7 +575,7 @@ export const CIStepOptionalConfig: React.FC<CIStepOptionalConfigProps> = props =
           })}
         </div>
       )}
-      {get(enableFields, 'spec.settings')
+      {get(enableFields, MapComponentFieldNames.SETTINGS)
```
Comment (mtran7) [inline src/modules/75-ci/components/PipelineSteps/CIStep/CIStepOptionalConfig.tsx:578]: used this reference so people in the future will add to the MapComponentFieldNames, read the comment, and go to the stageHelpers.ts file

**Review by vkbansal-harness — APPROVED** [review]

File: src/modules/10-common/components/MultiTypeMapInputSet/MultiTypeMapInputSet.tsx [hunk 3]
```diff
@@ -125,25 +146,43 @@ export const MultiTypeMapInputSet = (props: MultiTypeMapProps): React.ReactEleme
...
+      !isEqual(appliedInputSetValue, valueInCorrectFormat)
+    ) {
+      const newVal = getInitialValueInCorrectFormat(appliedInputSetValue)
+      setValue(newVal)
+      formikFieldValues = appliedInputSetValue
+    }
+  }, [appliedInputSetValue])
+
+  React.useEffect(() => {
+    let initialValue = formikFieldValues
```
Comment (mtran7) [inline src/modules/10-common/components/MultiTypeMapInputSet/MultiTypeMapInputSet.tsx:178]: will change to const to fix ESLint check

**Review by tan-nhu — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

**Pull request merged**